### ISSUE TITLE
repeated ctrl-z is queued

### ISSUE THREAD

**Issue by mqudsi** [neovim/neovim#8310]
- `nvim --version`: v0.3.0-1063-g51af911
- Vim (version: 8.0): yes 
- Operating system/version: WSL/17134.1
- Terminal name/version: conemu
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:set autowrite
" do something that will cause ctrl+z to take up a second or so, i.e. not happen instantly, 
" for example, saving a huge file from scratch on a slow device with fsync on, etc.
<ESC>
ctrl+z "spam this repeatedly:
ctrl+z
ctrl+z
```

Back in the terminal:
`fg`

### Actual behaviour
neovim restores and then executes the last-queued `ctrl+z` causing it it to immediately suspend itself.

### Expected behaviour
ctrl+z should not stack, or `:resume` should clear all pending `:suspend` commands (why is any input being persisted after a suspend and resume, actually?).

**Comment by huyvohcmc** [comment]
I can't reproduce the issue, when I `fg` to come back to nvim nothing weird happens.

**Comment by justinmk** [comment]
@huyvohcmc It's timing-dependent. To ensure that two `<c-z>` commands are in the input-queue, try this:

    :call nvim_input('<c-z><c-z>')

After `fg` from the shell, the second `<c-z>` is processed and user is immediately returned to the shell. Invoking `fg` again returns to Nvim.

**Comment by ZyX-I** [comment]
I do not think this is really a “low complexity” issue. Discarding sequential `<C-z>` is easy only until you take into account mappings: at least ones which do some stuff and `<C-z>` afterwards and aliases to `<C-z>`. I would close that as both “working as intended” and “can use mapping for a solution to a specific case” both apply to the issue.

(BTW, following OP saing about doing some heavy writing I can reproduce the same thing on Vim without using `feedkeys()`: could just write to my WebDAV mount.)

**Comment by mqudsi** [comment]
@justinmk is it closed as wontfix because it repros in vim?

**Comment by justinmk** [comment]
See zyx comment.

**Comment by mqudsi** [comment]
Right. I took that to mean that queuing of c-z is not a bug, and is working correctly. But suspend immediately after resume is still a bug.
